app/src/main/java/com/khuttun/notificationnotes/NotificationMgr.java

```diff
@@ -90,43 +85,158 @@ public void setGroupNotification(ArrayList<NotificationNote> notes)
         this.notificationManager.cancelAll();
     }
 
+
     private Notification createNotification(String title, String text)
     {
-        this.notificationBuilder.setStyle(new NotificationCompat.BigTextStyle().bigText(text));
-        this.notificationBuilder.setContentTitle(title);
-        this.notificationBuilder.setContentText(text);
-        return this.notificationBuilder.build();
+        NotificationCompat.Builder notificationBuilder = getNotificationBuilder();
+        notificationBuilder.setStyle(new NotificationCompat.BigTextStyle().bigText(text));
+        notificationBuilder.setContentTitle(title);
+        notificationBuilder.setContentText(text);
+        return notificationBuilder.build();
     }
 
-    private Notification createGroupNotification(ArrayList<NotificationNote> notes)
+
+    /**
+     * Creates a notification with the number of notes and the first line of each note.
+     * Visible for Android versions below Nougat (7.0).
+     * Is required for Android versions Nougat and beyond, but does not show.
+     */
+    private Notification createSummaryNotification(ArrayList<NotificationNote> notes)
     {
+        NotificationCompat.Builder notificationBuilder = getNotificationBuilder();
         NotificationCompat.InboxStyle style = new NotificationCompat.InboxStyle();
         for (int i = 0; i < notes.size(); ++i)
         {
             style.addLine(getGroupNotificationLine(notes.get(i)));
         }
-        this.notificationBuilder.setStyle(style);
-        this.notificationBuilder.setContentTitle(
+        notificationBuilder.setStyle(style);
+        notificationBuilder.setContentTitle(
                 this.context.getString(R.string.group_notif_title) + ": " + notes.size());
-        this.notificationBuilder.setContentText(getGroupNotificationLine(notes.get(0)));
-        return this.notificationBuilder.build();
+        notificationBuilder.setContentText(getGroupNotificationLine(notes.get(0)));
+        notificationBuilder.setGroup(GROUP_KEY);
+        notificationBuilder.setGroupSummary(true);
+        return notificationBuilder.build();
     }
 
+
     private CharSequence getGroupNotificationLine(NotificationNote note)
     {
         Spannable sp = new SpannableString(note.title + " " + note.text);
         sp.setSpan(new StyleSpan(Typeface.BOLD), 0, note.title.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
         return sp;
     }
+
+    private ArrayList<Notification> createGroupNotifications(ArrayList<NotificationNote> notes)
+    {
+        ArrayList<Notification> groupedNotes = new ArrayList<Notification>();
+        Notification summary = createSummaryNotification(notes);
+        groupedNotes.add(summary);
+        for (int i = 0; i < notes.size(); ++i)
+        {
+            NotificationNote note = notes.get(i);
+            Notification newNote = createGroupNotification(note.title, note.text);
+            groupedNotes.add(newNote);
+        }
+
+        if (groupAPI)
```

Comment:
Is this check necessary here? Is this function ever called if groupAPI is false?